contracts/src/PDPRecordKeeper.sol

```diff
@@ -35,13 +44,29 @@ contract PDPRecordKeeper is PDPListener {
         _;
     }
 
-    // Function to add a new event record
-    function receiveProofSetEvent(
-        uint256 proofSetId,
-        uint64 epoch,
-        PDPListener.OperationType operationType,
-        bytes calldata extraData
-    ) external onlyPDPService {
+    // Listener interface methods
+    function proofSetCreated(uint256 proofSetId, address creator) external onlyPDPService {
+        receiveProofSetEvent(proofSetId, OperationType.CREATE, abi.encode(creator));
+    }
+
+    function rootsAdded(uint256 proofSetId, uint256 firstAdded, PDPService.RootData[] memory rootData) external onlyPDPService {
+        receiveProofSetEvent(proofSetId, OperationType.ADD, abi.encode(firstAdded, rootData));
+    }
+
+    function rootsScheduledRemove(uint256 proofSetId, uint256[] memory rootIds) external onlyPDPService {
+        receiveProofSetEvent(proofSetId, OperationType.REMOVE, abi.encode(rootIds));
```

Comment:
Note the total delta was dropped from the event payload here, following our discussion about it re #46